kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Channel.java

```diff
@@ -1,168 +0,0 @@
-    Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = Maps.newHashMap();
-    sourceOffsets.forEach((k, v) -> offsetsToCommit.put(k, new OffsetAndMetadata(v.offset())));
-
-    List<ProducerRecord<String, byte[]>> recordList =
-        events.stream()
-            .map(
-                event -> {
-                  LOG.info("Sending event of type: {}", event.type().name());
-                  byte[] data = Event.encode(event);
-                  // key by producer ID to keep event order
-                  return new ProducerRecord<>(controlTopic, producerId, data);
-                })
-            .collect(toList());
-
-    synchronized (producer) {
-      producer.beginTransaction();
-      try {
-        recordList.forEach(producer::send);
-        if (!sourceOffsets.isEmpty()) {
-          // TODO: this doesn't fence zombies
-          producer.sendOffsetsToTransaction(
-              offsetsToCommit, new ConsumerGroupMetadata(controlGroupId));
-        }
-        producer.commitTransaction();
-      } catch (Exception e) {
-        try {
-          producer.abortTransaction();
-        } catch (Exception ex) {
-          LOG.warn("Error aborting producer transaction", ex);
-        }
-        throw e;
-      }
-    }
-  }
-
-  protected abstract boolean receive(Envelope envelope);
-
-  protected void consumeAvailable(Duration pollDuration) {
-    ConsumerRecords<String, byte[]> records = consumer.poll(pollDuration);
-    while (!records.isEmpty()) {
-      records.forEach(
-          record -> {
-            // the consumer stores the offsets that corresponds to the next record to consume,
-            // so increment the record offset by one
-            controlTopicOffsets.put(record.partition(), record.offset() + 1);
-
-            Event event = Event.decode(record.value());
-
-            if (event.groupId().equals(groupId)) {
-              LOG.debug("Received event of type: {}", event.type().name());
-              if (receive(new Envelope(event, record.partition(), record.offset()))) {
-                LOG.info("Handled event of type: {}", event.type().name());
-              }
-            }
-          });
-      records = consumer.poll(pollDuration);
-    }
-  }
-
-  protected Map<Integer, Long> controlTopicOffsets() {
-    return controlTopicOffsets;
-  }
-
-  protected void commitConsumerOffsets() {
-    Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = Maps.newHashMap();
-    controlTopicOffsets()
-        .forEach(
-            (k, v) ->
-                offsetsToCommit.put(new TopicPartition(controlTopic, k), new OffsetAndMetadata(v)));
-    consumer.commitSync(offsetsToCommit);
-  }
-
-  protected Admin admin() {
-    return admin;
-  }
-
-  public void start() {
-    consumer.subscribe(ImmutableList.of(controlTopic));
-
-    // initial poll with longer duration so the consumer will initialize...
-    consumeAvailable(Duration.ofMillis(1000));
```

Comment:
We had this logic previously to do a long poll the first time a consumer is started. 
We do not have this logic now. 
I haven't seen any issues with it but would like to know if I'm missing something. 

I'll do some tests locally as well to see if this is an issue.

Reply:
Looking at https://stackoverflow.com/questions/55067611/kafka-consumer-needs-a-long-poll-duration , https://chrzaszcz.dev/2019/06/16/kafka-consumer-poll/ makes it look like the first `poll` requires me time than subsequent polls to finish the handshake.    

Didn't spend too much time digging through Kafka codebase but I wonder if we can get into a situation where our consumer never initializes.    Seems possible from the look at KafkaConsumer / ConsumerNetworkClient

Reply:
Yes, I encountered situations where the client would never initialize, hence this long poll.

Reply:
Added back the long poll duration for the first poll only